### PR TITLE
Hide staff identities from applicants & partners

### LINKED ISSUE #3936: Applicants should not be able to see what staff member is lead on their application
Currently when a user submits an application, they can see who the staff lead is. In other cases we obscure the staff members name and it probably also makes sense to do so here as it's PII.

Screenshot 2024-05-22 at 11 21 05

### PR DESCRIPTION
**Pull request by wes-otf**
Fixes #3936. Hides staff identities and adds the new settings option of `HIDE_STAFF_IDENTITY` to make this option configurable. Also, in adding `is_org_faculty` I noticed `is_apply_staff_or_finance` which was only used in one place and felt redundant so I tossed it.

## Test Steps

**Testing w/ `HIDE_STAFF_IDENTITY` enabled**

 - [ ] As an applicant role, submit an application
 - [ ] View your application and ensure you cannot see the lead 
 - [ ] As a staff role, leave a comment on the submitted application
 - [ ] As the original applicant, ensure the staff comment appears as the organization name

**Testing w/ `HIDE_STAFF_IDENTITY` disabled**

 - [ ] As an applicant role, submit an application
 - [ ] View your application and ensure you can see the lead 
 - [ ] As a staff role, leave a comment on the submitted application
 - [ ] As the original applicant, ensure the staff comment appears as the staff member's name

### CODE REVIEW

File: hypha/apply/activity/templatetags/activity_tags.py [hunk 1]
```diff
@@ -27,10 +27,13 @@ def display_author(activity, user) -> str:
...
-        activity.user.is_apply_staff
-        or activity.user.is_finance
-        or activity.user.is_contracting
+    if settings.HIDE_STAFF_IDENTITY and (
+        (user.is_applicant or user.is_partner)
+        and (
+            activity.user.is_apply_staff
+            or activity.user.is_finance
+            or activity.user.is_contracting
+        )
```
Comment (frjo) [inline hypha/apply/activity/templatetags/activity_tags.py]: `activity.user.is_org_faculty`

File: docs/setup/administrators/configuration.md [hunk 2]
```diff
@@ -175,6 +175,12 @@ Staff e-mail domain. Used for OAUTH2 whitelist default value and staff account c
 
 ----
 
+Should staff identities be obscured (ie. comments will be ORG_LONG_NAME rather than "John Doe").
+
+    SUBMISSIONS_DRAFT_ACCESS_STAFF = env.bool('SUBMISSIONS_DRAFT_ACCESS_STAFF', False)
```
Comment (frjo) [inline docs/setup/administrators/configuration.md]: `HIDE_STAFF_IDENTITY`

File: hypha/settings/base.py [hunk 3]
```diff
@@ -115,6 +115,9 @@
 # Staff e-mail domain. Used for OAUTH2 whitelist default value and staff account creation.
 STAFF_EMAIL_DOMAINS = env.list("STAFF_EMAIL_DOMAINS", [])
 
+# Should staff identities be obscured (ie. comments will be ORG_LONG_NAME rather than John Doe).
```
Comment (frjo) [inline hypha/settings/base.py]: Maybe add that it is hidden from applicant/partner?

**Comment by wes-otf** [comment]
Drafting as I wanted to see if there was a way a migration could be created for this to make all previous `Lead updated...` activities to have the visibility of team? Looking at the tables I was struggling to see how I could correlate the `activity_event` table's `type` back to the `activity_activity` table to modify visibility. ie, all `activity_activity` items triggered by `UPDATE_LEAD` events would have their visibility changed to team. Digging more today but am open to any ideas! 

Will also be adding unit tests & looking at the project side of this

File: docs/setup/administrators/configuration.md (hunk 2 above)
Comment (wes-otf) [reply docs/setup/administrators/configuration.md]: gah my copy & pastes are getting me in trouble

**Comment by frjo** [comment]
Looks good!

We had staff named in some activities as well I think you mentioned. And the Lead name.

The list of reviews in the submission detail sidebar lists names as well I believe.

**Comment by wes-otf** [comment]
I rebased onto #3913 to be able to utilize the `recipient` email template context, so this is dependent on that.

All is just about set here, just need to figure out how to [migrate those previous activities ](https://github.com/HyphaApp/hypha/pull/3948#issuecomment-2135767198) & write some unit tests

File: hypha/apply/projects/templates/application_projects/vendor_success.html [hunk 4]
```diff
@@ -10,7 +10,7 @@
 
     <div class="wrapper wrapper--small">
         <h3>{% trans "Thank you for submitting your information." %}</h3>
-        <p>{% trans "Your {{ ORG_SHORT_NAME }} Programme Manager will be in touch with you if there are any issues." %}</p>
+        <p>{% blocktrans %}Your {{ ORG_SHORT_NAME }} Programme Manager will be in touch with you if there are any issues.{% endblocktrans %}</p>
```
Comment (wes-otf) [inline hypha/apply/projects/templates/application_projects/vendor_success.html:13]: Quick translation bug I found, unrelated to this PR.

**Comment by wes-otf** [comment]
This should be ready to be tested - as mentioned before it's rebased onto #3913 so some of the diff comes from that (see [here](https://github.com/HyphaApp/hypha/pull/3948/files/3041a889a071d761f020f20e95c8337d62445eb8..14142e79164311a471563ca5a81547ba106cd367) for commits just from this PR). Unit tests added to make sure the lead's name doesn't show anywhere in the project/submission detail views when identity hiding is enabled

**Comment by wes-otf** [comment]
@frjo this is ready to go on test

File: hypha/apply/users/models.py [hunk 5]
```diff
@@ -296,6 +292,10 @@ def is_finance_level_1(self):
             and not self.groups.filter(name=APPROVER_GROUP_NAME).exists()
         )
 
+    @cached_property
+    def is_org_faculty(self):
+        return self.is_apply_staff or self.is_finance or self.is_contracting
+
```
Comment (frjo) [inline hypha/apply/users/models.py:298]: Good generic name for this!

File: hypha/apply/users/forms.py [hunk 6]
```diff
@@ -164,7 +164,7 @@ class Meta:
     def __init__(self, *args, **kwargs):
         self.request = kwargs.pop("request", None)
         super().__init__(*args, **kwargs)
-        if not self.instance.is_apply_staff_or_finance:
+        if not (self.instance.is_apply_staff or self.instance.is_finance):
```
Comment (frjo) [inline hypha/apply/users/forms.py]: Can we not use the new `is_org_faculty` here?

File: hypha/settings/base.py [hunk 7]
```diff
@@ -120,6 +120,9 @@
 # Staff e-mail domain. Used for OAUTH2 whitelist default value and staff account creation.
 STAFF_EMAIL_DOMAINS = env.list("STAFF_EMAIL_DOMAINS", [])
 
+# Should staff identities be obscured from Applicants & Partners (ie. comments will be ORG_LONG_NAME rather than John Doe).
+HIDE_STAFF_IDENTITY = env.bool("HIDE_STAFF_IDENTITY", False)
+
```
Comment (frjo) [inline hypha/settings/base.py:125]: Good naming and explanation!

**Review by frjo — CHANGES_REQUESTED** [review]

File: hypha/apply/users/forms.py (hunk 6 above)
Comment (wes-otf) [reply hypha/apply/users/forms.py]: Yes! this makes more sense too as contracting will likely be using slack. I just did a blind substitution after removing `is_apply_staff_or_finance`

**Review by frjo — APPROVED** [review]

**Comment by wes-otf** [comment]
Also gonna toss this to the user testing team - they were the ones that expressed concern over this originally, wanna see if they can break it

**Comment by wes-otf** [comment]
@frjo can you push this to test when you get a sec? going to get it user tested so we can hopefully get it in soon

**Comment by frjo** [comment]
@wes-otf Updated test now.

**Pull request merged**